pkg/common/client/config.go

```diff
@@ -38,52 +39,202 @@ func MergeSaramaSettings(config *sarama.Config, saramaSettingsYamlString string)
 		config.Version = constants.ConfigKafkaVersionDefault
 
 		// Add Any Required Settings
-		UpdateSaramaConfig(config, config.ClientID, "", "")
+		InitSaramaConfig(config)
 	}
 
-	// Extract (Remove) The KafkaVersion From The Sarama Config YAML
-	saramaSettingsYamlString, kafkaVersion, err := extractKafkaVersion(saramaSettingsYamlString)
-	if err != nil {
-		return nil, fmt.Errorf("failed to extract KafkaVersion from Sarama Config YAML: err=%s : config=%+v", err, saramaSettingsYamlString)
+	if saramaSettingsYamlString != "" {
+		// Extract (Remove) The KafkaVersion From The Sarama Config YAML as we can't marshal it regularly
+		saramaSettingsYamlString, kafkaVersionInYaml, err := extractKafkaVersion(saramaSettingsYamlString)
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract KafkaVersion from Sarama Config YAML: err=%s : config=%+v", err, saramaSettingsYamlString)
+		}
+
+		// Extract (Remove) Any TLS.Config RootCAs & Set In Sarama.Config
+		saramaSettingsYamlString, certPool, err := extractRootCerts(saramaSettingsYamlString)
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract RootPEMs from Sarama Config YAML: err=%s : config=%+v", err, saramaSettingsYamlString)
+		}
+
+		// Unmarshall The Sarama Config Yaml Into The Provided Sarama.Config Object
+		err = yaml.Unmarshal([]byte(saramaSettingsYamlString), &config)
+		if err != nil {
+			return nil, fmt.Errorf("ConfigMap's sarama value could not be converted to a Sarama.Config struct: %s : %v", err, saramaSettingsYamlString)
+		}
+
+		// Override The Custom Parsed KafkaVersion, if it is specified in the YAML
+		if kafkaVersionInYaml != nil {
+			config.Version = *kafkaVersionInYaml
+		}
+
+		// Override Any Custom Parsed TLS.Config.RootCAs
+		if certPool != nil && len(certPool.Subjects()) > 0 {
+			config.Net.TLS.Config = &tls.Config{RootCAs: certPool}
+		}
 	}
 
-	// Extract (Remove) Any TLS.Config RootCAs & Set In Sarama.Config
-	saramaSettingsYamlString, certPool, err := extractRootCerts(saramaSettingsYamlString)
+	err := UpdateSaramaConfigWithKafkaAuthConfig(config, kafkaAuthCfg)
 	if err != nil {
-		return nil, fmt.Errorf("failed to extract RootPEMs from Sarama Config YAML: err=%s : config=%+v", err, saramaSettingsYamlString)
+		return nil, err
 	}
 
-	// Unmarshall The Sarama Config Yaml Into The Provided Sarama.Config Object
-	err = yaml.Unmarshal([]byte(saramaSettingsYamlString), &config)
-	if err != nil {
-		return nil, fmt.Errorf("ConfigMap's sarama value could not be converted to a Sarama.Config struct: %s : %v", err, saramaSettingsYamlString)
-	}
+	// Return Success
+	return config, nil
+}
 
-	// Override The Custom Parsed KafkaVersion
-	config.Version = kafkaVersion
+func InitSaramaConfig(config *sarama.Config) {
```

Comment:
1. I think the name isn't reflecting what's happening here as mentioned on the comment above. 
2. Would be good to have a comment on the function.

Reply:
As above, I will rename this to `UpdateConfigWithDefaults` and add comments.